pkg/gameservers/health.go

```diff
@@ -190,15 +190,61 @@ func (hc *HealthController) syncGameServer(key string) error {
 		return nil
 	}
 
-	hc.loggerForGameServer(gs).Info("Issue with GameServer pod, marking as GameServerStateUnhealthy")
-	gsCopy := gs.DeepCopy()
-	gsCopy.Status.State = agonesv1.GameServerStateUnhealthy
-
-	if _, err := hc.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(gsCopy); err != nil {
-		return errors.Wrapf(err, "error updating GameServer %s to unhealthy", gs.ObjectMeta.Name)
+	skip, err := hc.skipUnhealthy(gs)
+	if err != nil {
+		return err
 	}
 
-	hc.recorder.Event(gs, corev1.EventTypeWarning, string(gsCopy.Status.State), "Issue with Gameserver pod")
+	if !skip {
```

Comment:
This should be inverted. 

```
if skip {
  // The game server hasn't been marked ready, so it doesn't transition to unhealthy. 
  return nil
}
```

To leave the other code with one less level of indentation.

Reply:
Ah excellent suggestion!